### PR TITLE
Gap 1977 update grant subscription to handle sub

### PR DESCRIPTION
**Pull request by dylanwrightCO**
Ticket # https://technologyprogramme.atlassian.net/browse/GAP-1977

### CODE REVIEW

File: src/subscription/subscription.controller.ts [hunk 1]
```diff
@@ -36,26 +36,30 @@ export class SubscriptionController {
...
+
+        return subscription
+            ? subscription
+            : this.subscriptionService.findByEmailAndGrantId(id, grantId);
     }
 
-    @Delete('users/:plainTextEmailAddress/grants/:grantId')
-    async deleteByEmailAndGrantId(
-        @Param('plainTextEmailAddress') plainTextEmailAddress: string,
+    @Delete('users/:id/grants/:grantId')
```
Comment (john-tco) [inline src/subscription/subscription.controller.ts:54]: This endpoint is currently consumed by the front-end unsub page `gap-find-web/pages/unsubscribe/[id].tsx` Needs updated there too

File: src/user/user.service.ts [hunk 2]
```diff
@@ -14,11 +14,22 @@ export class UserService {
...
+                    sub,
+                },
+            });
+        } else {
+            foundUser = await this.userRepository.findOne({
+                where: {
+                    hashedEmailAddress,
+                },
+            });
+        }
```
Comment (john-tco) [inline src/user/user.service.ts:32]:
```suggestion
	       const query = sub ? { sub } : { hashedEmailAddress };
        	const foundUser = await this.userRepository.findOne({
	            where: query,
        	});
```

File: src/subscription/subscription.service.ts [hunk 3]
```diff
@@ -105,4 +129,23 @@ export class SubscriptionService {
 
         return deleteResult;
     }
+
+    async deleteBySubAndGrantId(
+        sub: string,
+        contentfulGrantSubscriptionId: string,
+    ): Promise<DeleteResult> {
```
Comment (john-tco) [inline src/subscription/subscription.service.ts:136]: This type can be inferred
```suggestion
    ) {
```

File: src/user/user.service.ts (hunk 2 above)
Comment (dylanwrightCO) [reply src/user/user.service.ts]: Nice suggestion!

**Review by john-tco — APPROVED** [review]

**Pull request merged**